src/main/java/com/example/springboot/repository/UserRepository.java

```diff
@@ -0,0 +1,9 @@
+package com.example.springboot.repository;
+
+import com.example.springboot.model.User;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface UserRepository extends JpaRepository<User, Long> {
+    Optional<User> findUsersByPhoneNumber(String phoneNumber);
```

Comment:
see checklist - you should join fetch roles along with user in this method, and also in `getById()`